backend/src/transit_route.rs

```diff
@@ -0,0 +1,125 @@
+use std::collections::hash_map::Entry;
+use std::collections::{BinaryHeap, HashMap};
+use std::time::Duration;
+
+use anyhow::{bail, Result};
+use chrono::NaiveTime;
+use geo::LineString;
+use geojson::{Feature, GeoJson, Geometry};
+use utils::PriorityQueueItem;
+
+use crate::costs::cost;
+use crate::graph::{Graph, IntersectionID, Mode, RoadID};
+use crate::gtfs::{StopID, TripID};
+
+pub fn route(graph: &Graph, start: IntersectionID, end: IntersectionID) -> Result<String> {
+    // TODO We'll need a start time too (and a day of the week)
+    let start_time = NaiveTime::from_hms_opt(7, 0, 0).unwrap();
+    if start == end {
+        bail!("start = end");
+    }
+
+    // Dijkstra implementation just for walking
+
+    // TODO stops are associated with roads, so the steps using transit are going to look a little
```

Comment:
The most confusing thing right now is the graph. What're the nodes and edges being searched? Right now, the nodes are (street) intersections and the edges (`PathStep` enum at the bottom of this file) are either the roads in between (bidirectional, because walking) or taking a bus. But that means when we get to the other end of a bus edge, we just have to appear at both intersection nodes. (Or we could do one and let the walking edge happen next naturally, actually.)

In A/B Street ages ago, I was more precise with this, and the graph was much more complicated, because the turning movements between roads was also represented. It was possible to appear at a precise bus stop location (on the correct side of the road). Not sure what's worth doing here yet.